packages/vjudge/src/providers/codeforces.ts

```diff
@@ -114,6 +114,10 @@ export default class CodeforcesProvider extends BasicFetcher implements IBasicPr
         if (await this.loggedIn) return true;
         logger.info('retry normal login');
         const [csrf, ftaa, bfaa] = await this.getCsrfToken('/enter');
+        const { header } = await this.get('/enter');
+        if (header['set-cookie']) {
```

Comment:
maybe this can be done in `getCsrfToken` call to preven duplicate fetches? (also return `resp.header` in `fetch.ts#html`)